cookbook/testing/integration-test-scrolling.md

```diff
@@ -0,0 +1,208 @@
+keep this recipe focused on testing, we'll use the app we created in the
+[Working with long lists](/cookbook/lists/long-lists/) recipe. If you're unsure
+of how to work with lists of content, please see that recipe for an 
+introduction.
+
+As we did in the [Introduction to integration
+testing](/cookbook/testing/integration-test-introduction/) recipe, we'll also
+add Keys to the widgets we want to interact with inside our integration tests.
+
+```dart
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+void main() {
+  runApp(MyApp(
+    items: List<String>.generate(10000, (i) => "Item $i"),
+  ));
+}
+
+class MyApp extends StatelessWidget {
+  final List<String> items;
+
+  MyApp({Key key, @required this.items}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final title = 'Long List';
+
+    return MaterialApp(
+      title: title,
+      home: Scaffold(
+        appBar: AppBar(
+          title: Text(title),
+        ),
+        body: ListView.builder(
+          // Add a key to the ListView. This allows us to find the list and
+          // scroll through it in our tests
+          key: Key('long_list'),
+          itemCount: items.length,
+          itemBuilder: (context, index) {
+            return ListTile(
+              title: Text(
+                '${items[index]}',
+                // Add a key to the Text Widget for each item. This allows
+                // us to look for a particular item in the list and verify the
+                // text is correct
+                key: Key('item_${index}_text'),
+              ),
+            );
+          },
+        ),
+      ),
+    );
+  }
+}
+```
+
+### 2. Instrument the app
+
+Next, we'll need to create an instrumented version of our app. This code lives
+in a file called `test_driver/app.dart`.
+
+<!-- skip -->
+```dart
+import 'package:flutter_driver/driver_extension.dart';
+import 'package:scrollable_app/main.dart' as app;
+
+void main() {
+  // This line enables the extension
+  enableFlutterDriverExtension();
+
+  // Call the `main()` function of your app or call `runApp` with any widget you 
+  // are interested in testing.
+  app.main();
+}
+```
+
+### 3. Write a test that scrolls through the list
+
+Now, we can write our test! In this example, we need to scroll through the list
+of items and verify that a particular item exists in the list. The
+[`FlutterDriver`](https://docs.flutter.io/flutter/flutter_driver/FlutterDriver-class.html)
+class provides three methods for scrolling through lists:
+
+  - The
+  [`scroll`](https://docs.flutter.io/flutter/flutter_driver/FlutterDriver/scroll.html)
+  method allows us to scroll through a specific list by a given amount. 
+  - The
+  [`scrollIntoView`](https://docs.flutter.io/flutter/flutter_driver/FlutterDriver/scrollIntoView.html)
+  method finds a specific Widget that's currently displayed on screen and will
```

Comment:
nit: "currently displayed on screen" may not be accurate. It may not actually be visible on the screen. It might be rendered offscreen (e.g. clipped by a scrollable viewport). However, the widget we're looking for must be instantiated and rendered into the render object hierarchy, even if its pixels are not painted on screen.

This method is useful when you know that something is eagerly instantiated and all you want is shift to a position such that it's visible. Perhaps you are scrolling a `SingleChildScrollView`. Or perhaps you are testing a scrollable container that renders widgets on-demand (e.g. [ListView.builder](https://docs.flutter.io/flutter/widgets/ListView/ListView.builder.html)) and you expect that at a certain position a certain widget must currently be rendered.

Contrast this to `scrollUntilVisible`, which will scroll "hoping" that eventually the widget we are looking for will be rendered. While `scrollIntoView` will fail instantly if it's not already there, `scrollUntilVisible` may keep scrolling hopelessly until it times out (especially if the list is indefinite).

**Suggestion**: perhaps instead of "displayed" or "visible", we say "rendered", which strongly implies `RenderObject`. A paragraph explaining the difference between being "rendered" and being "visible" might help readers better understand how these methods work (and the framework in general; scrolling is complicated :sweat_smile:).

Reply:
Good call -- I'll definitely add some more info here since, as you can tell, I was a bit confused by looking at the API docs on exactly how that was working :P

Would it be ok to ping you for a review once I put it in? I definitely want to make sure everything is technically correct.

Reply:
Of course! Also, if you can come up with a better wording for our dartdocs, please feel free to send a PR to improve it.